### PR TITLE
More verbose error for wrong config file

### PR DESCRIPTION
**Pull request by rzadp**

### CODE REVIEW

File: waffle-compiler/src/config/inputToConfig.ts [hunk 1]
```diff
@@ -46,19 +46,20 @@ export function inputToConfig(input: InputConfig) {
...
-    throw new TypeError('Invalid config');
+  const results = [
+    checkSourceDirectory(config.sourceDirectory) || 'source directory',
+    checkOutputDirectory(config.outputDirectory) || 'output directory',
+    checkNodeModulesDirectory(config.nodeModulesDirectory) || 'node modules directory',
+    checkCompilerType(config.compilerType) || 'compiler type',
+    checkCompilerVersion(config.compilerVersion) || 'compiler version',
+    checkCompilerAllowedPaths(config.compilerAllowedPaths) || 'compiler allowed paths',
+    checkCompilerOptions(config.compilerOptions) || 'compiler options',
+    checkOutputHumanReadableAbi(config.outputHumanReadableAbi) || 'output human readable ABI',
```
Comment (sz-piotr) [inline waffle-compiler/src/config/inputToConfig.ts:57]: I think we should just use the proper keys, so `outputHumanReadableAbi`. `Invalid config. Check your output human readable ABI` is in my opinion inferior to `'Invalid config. Check the value of "outputHumanReadableAbi"`

Comment (rzadp) [reply waffle-compiler/src/config/inputToConfig.ts]: ✅

File: waffle-compiler/src/config/inputToConfig.ts [hunk 2]
```diff
@@ -46,20 +46,21 @@ export function inputToConfig(input: InputConfig) {
...
+
+  checkConfigProperty('sourceDirectory', checkSourceDirectory);
+  checkConfigProperty('outputDirectory', checkOutputDirectory);
+  checkConfigProperty('nodeModulesDirectory', checkNodeModulesDirectory);
+  checkConfigProperty('compilerType', checkCompilerType);
+  checkConfigProperty('compilerVersion', checkCompilerVersion);
+  checkConfigProperty('compilerAllowedPaths', checkCompilerAllowedPaths);
+  checkConfigProperty('compilerOptions', checkCompilerOptions);
+  checkConfigProperty('outputHumanReadableAbi', checkOutputHumanReadableAbi);
+  checkConfigProperty('outputType', checkOutputType);
```
Comment (marekkirejczyk) [inline waffle-compiler/src/config/inputToConfig.ts:63]: No tests? :P

Comment (rzadp) [reply waffle-compiler/src/config/inputToConfig.ts:63]: Seems covered [here](https://github.com/EthWorks/Waffle/blob/f373e7c020921b3fec67c341add0e0ba68e1a63d/waffle-compiler/test/config.ts#L4)

**Pull request merged**